connectors/sources/mysql.py

```diff
@@ -224,16 +218,27 @@ async def ping(self):
             if self.ssl_enabled
             else None,
         }
-        logger.info("Pinging MySQL...")
-        if self.connection_pool is None:
-            self.connection_pool = await aiomysql.create_pool(**connection_string)
+
         try:
-            async with self.connection_pool.acquire() as connection:
-                await connection.ping()
-                logger.info("Successfully connected to the MySQL Server.")
-        except Exception:
-            logger.exception("Error while connecting to the MySQL Server.")
-            raise
+            connection_pool = await aiomysql.create_pool(**connection_string)
+            yield connection_pool
+        finally:
+            if connection_pool:
+                connection_pool.close()
+                await connection_pool.wait_closed()
+
+    async def ping(self):
+        """Verify the connection with MySQL server"""
+
+        logger.info("Pinging MySQL...")
+        async with self.with_connection_pool() as connection_pool:
+            try:
+                async with connection_pool.acquire() as connection:
```

Comment:
It would be nice to include `acquire` in the context manager so we don't need this extra step here

Reply:
As discussed, I did not want to make a functional change here - it was implemented by Crest this way (1 connection pool -> X connections). Intuitively in positive cases it would not change anything, but in case retry policy kicks in there might be some impact. Probably minor, but at this point I just don't feel like taking the risk, this change can be done in a separate PR later.